src/test/java/org/cactoos/list/ListEnvelopeTest.java

```diff
@@ -73,97 +61,113 @@ public void returnsListIteratorWithSupportedSet() {
+        list.remove(0);
         new Assertion<>(
-            "subList.listIterator().set() must throw exception",
-            () -> {
-                final ListIterator<String> iterator = new StringList("one", "two", "three")
-                    .subList(0, 2)
-                    .listIterator(0);
-                iterator.next();
-                iterator.set("zero");
-                return new Object();
-            },
-            new Throws<>(
-                new MatcherOf<>(
-                    (String msg) -> msg.equals(
-                        "List Iterator is read-only and doesn't allow rewriting items"
-                    )
-                ),
-                UnsupportedOperationException.class
-            )
+            "must be empty",
+            list,
+            new IsEmptyCollection<>()
         ).affirm();
     }
 
-    @Test(expected = UnsupportedOperationException.class)
-    public void subListReturnsListIteratorWithUnsupportedAdd() {
+    @Test
+    public void indexOfIsDelegated() {
         final ListEnvelope<String> list = new StringList("one");
-        final ListIterator<String> iterator = list.subList(0, 1)
-            .listIterator();
-        iterator.next();
-        iterator.add("two");
+        new Assertion<>(
+            "must return corrent index",
+            list.indexOf("one"),
+            new IsEqual<>(0)
+        ).affirm();
     }
 
-    @Test(expected = UnsupportedOperationException.class)
-    public void removeIsNotSupported() {
+    @Test
+    public void addAllIsDelegated() {
         final ListEnvelope<String> list = new StringList("one");
-        list.remove(0);
+        list.addAll(0, new StringList("two"));
+        new Assertion<>(
+            "must be empty",
+            list,
+            new IsEqual<>(new ListOf<>("two", "one"))
+        ).affirm();
     }
 
-    @Test(expected = UnsupportedOperationException.class)
-    public void setIsNotSupported() {
+    @Test
+    public void setIsDelegatedToTheOriginal() {
         final ListEnvelope<String> list = new StringList("one");
         list.set(0, "zero");
+        new Assertion<>(
+            "value must be changed",
+            list,
+            new HasValues<>("zero")
+        ).affirm();
     }
 
-    @Test(expected = UnsupportedOperationException.class)
-    public void addIsNotSupported() {
+    @Test
+    public void addIsDelegated() {
         final ListEnvelope<String> list = new StringList("one");
         list.add("two");
+        new Assertion<>(
+            "value must be added",
+            list,
+            new HasValues<>("one", "two")
+        ).affirm();
     }
 
-    @Test(expected = UnsupportedOperationException.class)
-    public void returnsSubListWithUnsupportedRemove() {
+    public void returnsSubListWithRemove() {
         final ListEnvelope<String> list = new StringList("one");
         list.subList(0, 1).remove(0);
+        new Assertion<>(
+            "must be empty after removal",
```

Comment:
@andreoss Please, add more detail in this assertion message (as explained before).